demo-app/src/main/java/io/opentelemetry/android/demo/ui/shop/cart/CheckoutInfo.kt

```diff
@@ -0,0 +1,150 @@
+            InfoField(
+                value = value,
+                onValueChange = onValueChange,
+                label = label,
+                keyboardType = when (label) {
+                    "Zip Code", "Credit Card Number", "Month", "Year", "CVV" -> KeyboardType.Number
+                    else -> KeyboardType.Text
+                },
+                isPassword = label == "CVV"
+            )
+        }
+    }
+}
+
+@Composable
+fun InfoScreen() {
+    var shippingInfo by remember { mutableStateOf(ShippingInfo()) }
+    var paymentInfo by remember { mutableStateOf(PaymentInfo()) }
+
+    val focusManager = LocalFocusManager.current
+
+
+    val canProceed = shippingInfo.allFieldsNotBlank() && paymentInfo.allFieldsNotBlank()
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp)
+            .background(Color.White)
+            .clickable { focusManager.clearFocus() }
+            .verticalScroll(rememberScrollState())
+    ) {
+        SectionHeader(title = "Shipping Address")
+
+        InfoFieldsSection(
+            fields = listOf(
+                Triple("E-mail Address", shippingInfo.email) { shippingInfo = shippingInfo.copy(email = it) },
+                Triple("Street Address", shippingInfo.streetAddress) { shippingInfo = shippingInfo.copy(streetAddress = it) },
+                Triple("Zip Code", shippingInfo.zipCode) { shippingInfo = shippingInfo.copy(zipCode = it) },
+                Triple("City", shippingInfo.city) { shippingInfo = shippingInfo.copy(city = it) },
+                Triple("State", shippingInfo.state) { shippingInfo = shippingInfo.copy(state = it) },
+                Triple("Country", shippingInfo.country) { shippingInfo = shippingInfo.copy(country = it) }
+            )
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        SectionHeader(title = "Payment Method")
+
+        InfoFieldsSection(
+            fields = listOf(
+                Triple("Credit Card Number", paymentInfo.creditCardNumber) { paymentInfo = paymentInfo.copy(creditCardNumber = it) },
+                Triple("Month", paymentInfo.expiryMonth) { paymentInfo = paymentInfo.copy(expiryMonth = it) },
+                Triple("Year", paymentInfo.expiryYear) { paymentInfo = paymentInfo.copy(expiryYear = it) },
+                Triple("CVV", paymentInfo.cvv) { paymentInfo = paymentInfo.copy(cvv = it) }
+            )
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        Button(
+            onClick = { /*TODO Handle*/ },
+            modifier = Modifier.fillMaxWidth(),
+            enabled = canProceed
+        ) {
+            Text("Proceed")
+        }
+    }
+}
+
+fun <T : Any> T.allFieldsNotBlank(): Boolean {
+    return this::class.memberProperties.all { property ->
+        property.isAccessible = true
+        val value = property.call(this)
+        value is String && value.isNotBlank()
+    }
+}
```

Comment:
I think we should avoid this reflection and instead build explicit methods on each of the data classes. Something like this:

```
fun isComplete(): Boolean {
    return arrayOf(email, streetAddress, zipCode, city, state, country)
        .all { it.isNotBlank() }
}
```

The reflection feels a bit forced, and it modifies the permission/accessibility in an unwanted way. It also limits the ability add an optional field (like address line 2 or whatever) down the road.

Reply:
also if minified is enabled, the methods might be removed unless its added in the proguard file